app/controllers/api/person_escort_records_controller.rb

```diff
@@ -11,5 +18,22 @@ def assessment_class
     def assessment_serializer
       PersonEscortRecordSerializer
     end
+
+    def create_confirmation_event
+      now = Time.zone.now.iso8601
+
+      event_attributes = {
+        eventable: assessment,
+        occurred_at: now,
+        recorded_at: now,
+        notes: 'Automatically generated event',
+        details: {
+          confirmed_at: assessment.confirmed_at.iso8601,
+        },
+        created_by: created_by,
```

Comment:
I didn't add supplier id, as seems optional on events, and I don't think we need it here. It might be tricky trying to get a supplier at this point and will add unnecessary queries.

Reply:
I think(?) suppliers will create PERs - and so might be useful to capture which supplier confirmed the PER. You could get the supplier id directly in the controller (method `info_for_paper_trail` in ApplicationController)

Reply:
Bit late to this party, but the `created_by` will be the user if it's come from the frontend or the doorkeeper application owner name if not, so it will say which supplier did it if directly through the API.